src/overlays/actors/ovl_En_Tubo_Trap/z_en_tubo_trap.c

```diff
@@ -21,26 +27,317 @@ const ActorInit En_Tubo_Trap_InitVars = {
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Tubo_Trap_0x809307E0/func_8093089C.asm")
+    if (fabsf(this->actor.world.pos.y - this->targetY) < 10.0f) {
+        this->actor.speedXZ = 10.0f;
+        this->actor.world.rot.y = this->actor.yawTowardsPlayer;
+        this->actionFunc = &func_809311C4;
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Tubo_Trap_0x809307E0/func_809308F4.asm")
+// flying at link
+void func_809311C4(EnTuboTrap *this, GlobalContext *globalCtx) {
+    f32 dX = this->originPos.x - this->actor.world.pos.x;
+    f32 dY = this->originPos.y - this->actor.world.pos.y;
+    f32 dZ = this->originPos.z - this->actor.world.pos.z;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Tubo_Trap_0x809307E0/func_80930B60.asm")
+    func_800B8EC8(&this->actor, 0x3037U); // play flying sfx
+    
+    if ((SQ(dX) + SQ(dY) + SQ(dZ) > 57600.0f)) { //240.0f ^ 2
+        Math_SmoothScaleMaxF(&this->actor.gravity, -3.0f, 0.2f, 0.5f);
+    }
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Tubo_Trap_0x809307E0/func_80930DDC.asm")
+    this->actor.shape.rot.y += 0x1388;
+    func_80930DDC(this, globalCtx); // "handle impact"
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Tubo_Trap_0x809307E0/func_80931004.asm")
+void EnTuboTrap_Update(Actor *thisx, GlobalContext *globalCtx) {
+    EnTuboTrap* this = THIS;
+    s32 padding;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Tubo_Trap_0x809307E0/func_80931138.asm")
+    this->actionFunc(this, globalCtx);
+    Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+    func_800B78B8(globalCtx, &this->actor, 12.0f, 10.0f, 20.0f, 0x1F);
+    Actor_SetHeight(&this->actor, 0.0f);
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Tubo_Trap_0x809307E0/func_809311C4.asm")
+    if (this->actor.projectedPos.z < 811.0f) {
+        if (this->actor.projectedPos.z > 300.0f) {
+            this->actor.shape.shadowAlpha = (u8) ( (0x32B - (s32) this->actor.projectedPos.z) >> 1);
+            this->actor.shape.shadowDraw = func_800B3FC0;
+        } else if (this->actor.projectedPos.z > -10.0f) {
+            this->actor.shape.shadowAlpha = 0xFF;
+            this->actor.shape.shadowDraw = func_800B3FC0;
+        } else{
+            this->actor.shape.shadowDraw = NULL;
+        }
+    } else {
+        this->actor.shape.shadowDraw = NULL;
+    }
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Tubo_Trap_0x809307E0/EnTuboTrap_Update.asm")
+    Collision_CylinderMoveToActor(&this->actor, &this->collider);
+    Collision_AddAC(globalCtx, &globalCtx->colCheckCtx, &this->collider.base);
+    Collision_AddAT(globalCtx, &globalCtx->colCheckCtx, &this->collider.base);
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Tubo_Trap_0x809307E0/EnTuboTrap_Draw.asm")
+void EnTuboTrap_Draw(Actor *thisx, GlobalContext *globalCtx) {
+    //  Gfx_DrawDListOpa with a display list
+    func_800BDFC0(globalCtx, &D_05017EA0);
```

Comment:
Should be
```suggestion
    func_800BDFC0(globalCtx, D_05017EA0);
```
but only after `D_05017EA0` is a `Gfx[]`